### PR TITLE
AD9910: Write correct number of bits to POW register

### PR DESCRIPTION
**Pull request by pmldrmota**
# ARTIQ Pull Request

## Description of Changes
Write correct number of bits to POW register in `set_pow` to avoid symptom #1141. 
Moreover, return type hints resolve occasional compiler type inference issues.

### Related Issue

Prevents manifestation of #1141

## Type of Changes

|   | Type |
| ------------- | ------------- |
| ✓  | :bug: Bug fix  |

### All Pull Requests

- [x] Use correct spelling and grammar.

### Code Changes

- [x] Run `flake8` to check code style (follow PEP-8 style). `flake8` has issues with parsing Migen/gateware code, ignore as necessary.
- [x] Test your changes or have someone test them. Mention what was tested and how.
- [x] Add and check docstrings and comments
- [x] Check, test, and update the [unittests in /artiq/test/](../artiq/test/)

### Documentation Changes

- [x] Check, test, and update the documentation in [doc/](../doc/). Build documentation (`cd doc/manual/; make html`) to ensure no errors.

### Git Logistics

- [x] Split your contribution into logically separate changes (`git rebase --interactive`). Merge/squash/fixup commits that just fix or amend previous commits. Remove unintended changes & cleanup. See [tutorial](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase).
- [x] Write short & meaningful commit messages. Review each commit for messages (`git show`). Format:
  ```
  topic: description. < 50 characters total.
  
  Longer description. < 70 characters per line
  ```

### Licensing

See [copyright & licensing for more info](https://github.com/m-labs/artiq/blob/master/CONTRIBUTING.rst#copyright-and-sign-off).
ARTIQ files that do not contain a license header are copyrighted by M-Labs Limited and are licensed under LGPLv3+.

### CODE REVIEW

**Comment by jordens** [comment]
If you want to get this in, you have a much better chance by splitting this PR.
The type hints and the POW fix can go in. But I'd like to know whether a fixed delay instead of `break_realtime` will also fix the underflow issue. If yes, that's preferrable.

File: artiq/coredevice/ad9910.py [hunk 1]
```diff
@@ -222,6 +223,20 @@ def set_phase_mode(self, phase_mode):
...
+
+        :param addr: Register address
+        :param data: Data to be written
+        """
+        self.bus.set_config_mu(urukul.SPI_CONFIG, 8,
+                               urukul.SPIT_DDS_WR, self.chip_select)
+        self.bus.write(addr << 24)
+        self.bus.set_config_mu(urukul.SPI_CONFIG | spi.SPI_END, 16,
+                               urukul.SPIT_DDS_WR, self.chip_select)
+        self.bus.write(data)
```
Comment (jordens) [inline artiq/coredevice/ad9910.py:238]: 1. Did you actually test this? The `data` write should be MSB aligned.

2. This should be done in one xfer. Could you test the following?

```suggestion
        self.bus.set_config_mu(urukul.SPI_CONFIG | spi.SPI_END, 24,
                               urukul.SPIT_DDS_WR, self.chip_select)
        self.bus.write((addr << 24) | (data << 8))
```

**Comment by pmldrmota** [comment]
Tested using

    class RW16Test(EnvExperiment):
        def build(self):
            self.setattr_device("core")
            self.dev = self.get_device("urukul2_ch0")

        def run(self):
            self.read_write16()

        @kernel
        def read_write16(self):
            self.core.break_realtime()
            self.dev.cpld.init()
            self.dev.init()
            write = 0x1234
            self.dev.write16(_AD9910_REG_POW, write)
            self.dev.cpld.io_update.pulse_mu(8)
            read = self.dev.read16(_AD9910_REG_POW)
            self.print_hex(read)

        @rpc(flags={"async"})
        def print_hex(self, x):
            print(hex(x))

which outputs 0x1234 as expected.

**Comment by jordens** [comment]
I meant a functional test showing that the value has the correct effect: you'd e.g. toggle between 0x8000 and 0 and look on a scope.
But since the old code apparently didn't see that either, this is unlikely to regress.

**Pull request merged**